differ/differDriver.go

```diff
@@ -21,12 +21,44 @@ import (
 	"xdcrDiffer/base"
 	fdp "xdcrDiffer/fileDescriptorPool"
 	"xdcrDiffer/utils"
+
+	"github.com/couchbase/goxdcr/metadata"
+	"github.com/couchbase/goxdcr/service_def"
 )
 
 // For each ColID, the keys that have diffs
 type DiffKeysMap map[uint32][]string
 type MigrationHintMap map[string][]uint32
 
+type pruningWindow struct {
+	duration time.Duration
+	lock     sync.RWMutex
+}
+
+var sourcePruningWindow *pruningWindow = &pruningWindow{}
+
+func (p *pruningWindow) set(svc service_def.BucketTopologySvc, spec *metadata.ReplicationSpecification) error {
+	notificationCh, err := svc.SubscribeToLocalBucketFeed(spec, "")
+	if err != nil {
+		fmt.Printf("Failed to fetch LocalBucketFeed. err=%v\n", err)
+		return err
+	}
+	defer svc.UnSubscribeLocalBucketFeed(spec, "")
+	latestNotification := <-notificationCh
+	defer latestNotification.Recycle()
+	pruningWindow := latestNotification.GetVersionPruningWindowHrs()
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	p.duration = time.Duration(uint32(pruningWindow)*60*60) * time.Second
```

Comment:
Simply do
`time.Duration(uint32(pruningWindow)) * time.Hour`

Reply:
But shouldn't it in seconds for the comparison `xdcrBase.CasDuration(value1, cas) >= pruningWindow` ?

Reply:
@nelio2k I still think the pruning window should be kept in seconds 
This is how pruning window is used in goxdcr. We keep it in seconds here.
`if crMeta.NeedToUpdateHlv(meta, xmem.config.vbMaxCas[req.VBucket], time.Duration(atomic.LoadUint32(&xmem.config.hlvPruningWindowSec))*time.Second)`